src/runtime/rpc/network.cpp

```diff
@@ -442,6 +449,21 @@ void rpc_session::start_negotiation()
         }
 
         auth_negotiation();
+    } else {
+        // set negotiation success if auth is disabled
+        complete_negotiation(true);
+    }
+}
+
+void rpc_session::complete_negotiation(bool succ)
+{
+    if (succ) {
+        if (is_client()) {
+            set_connected();
+            on_send_completed();
+        }
```

Comment:
wrap as a function `on_success` would be better, it has the same level to `on_failure `